Framework/Algorithms/src/MonteCarloAbsorption.cpp

```diff
@@ -220,6 +213,72 @@ std::map<std::string, std::string> MonteCarloAbsorption::validateInputs() {
   return issues;
 }
 
+/**
+ * Factory method to return an instance of the required interaction volume
+ * class
+ * @param sample A reference to the object defining details of the sample
+ * @param maxScatterPtAttempts The maximum number of tries to generate a random
+ * point within the object
+ * @param pointsIn Where to generate the scattering point in
+ * @return a pointer to an MCAbsorptionStrategy object
+ */
+std::shared_ptr<IMCInteractionVolume>
+MonteCarloAbsorption::createInteractionVolume(
+    const API::Sample &sample, const size_t maxScatterPtAttempts,
+    const MCInteractionVolume::ScatteringPointVicinity pointsIn) {
+  auto interactionVol = std::make_shared<MCInteractionVolume>(
+      sample, maxScatterPtAttempts, pointsIn);
+  return interactionVol;
+}
+
+/**
+ * Factory method to return an instance of the required absorption strategy
+ * class
+ * @param interactionVol The interaction volume object to inject into the
+ * strategy
+ * @param beamProfile A reference to the object the beam profile
+ * @param EMode The energy mode of the instrument
+ * @param nevents The number of Monte Carlo events used in the simulation
+ * @param maxScatterPtAttempts The maximum number of tries to generate a random
+ * point within the object
+ * @param regenerateTracksForEachLambda Whether to resimulate tracks for each
+ * wavelength point or not
+ * @return a pointer to an MCAbsorptionStrategy object
+ */
+std::shared_ptr<IMCAbsorptionStrategy> MonteCarloAbsorption::createStrategy(
+    IMCInteractionVolume &interactionVol, const IBeamProfile &beamProfile,
+    Kernel::DeltaEMode::Type EMode, const size_t nevents,
+    const size_t maxScatterPtAttempts,
+    const bool regenerateTracksForEachLambda) {
+  auto MCAbs = std::make_shared<MCAbsorptionStrategy>(
+      interactionVol, beamProfile, EMode, nevents, maxScatterPtAttempts,
+      regenerateTracksForEachLambda);
+  return MCAbs;
+}
+
+/**
+ * Factory method to return an instance of the required SparseInstrument class
+ * @return a pointer to an SparseInstrument object
```

Comment:
This docstring appears to be missing the parameters

Reply:
Thanks, yes funny the Doxygen job doesn't pick that up. I'll insert the extra lines